NBitcoin/Protocol/Payloads/CompactFilterHeadersPayload.cs

```diff
@@ -0,0 +1,95 @@
+#if !NOSOCKET
+using NBitcoin.Crypto;
+using NBitcoin.Protocol.Behaviors;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NBitcoin.Protocol
+{
+	[Payload("cfheaders")]
+	public class CompactFilterHeadersPayload: Payload, IBitcoinSerializable
+	{
+		private byte _filterType = 0;
+		private uint256 _stopHash = new uint256();
+		private uint256 _previousFilterHeader = new uint256();
+		private VarInt _filterHashesLength = new VarInt(0);
+		private uint256[] _filterHashes = { };
+
+		public CompactFilterHeadersPayload(FilterType filterType, uint256 stopHash, uint256 previousFilterHeader, byte[] filterHashes)
+		{
+			if (filterType != FilterType.Basic /*&& filterType != FilterType.Extended*/) //Extended filters removed
+				throw new ArgumentException($"'{filterType}' is not a valid value. Try with Basic.", nameof(filterType));
+			if (stopHash == null)
+				throw new ArgumentException(nameof(stopHash));
+			if (previousFilterHeader == null)
+				throw new ArgumentException(nameof(previousFilterHeader));
+			if (filterHashes == null)
+				throw new ArgumentException(nameof(filterHashes));
+
+
+			FilterType = filterType;
+			_stopHash = stopHash;
+			_previousFilterHeader = previousFilterHeader;
+			_filterHashes = GetHashes(filterHashes);
+		}
+
+		public CompactFilterHeadersPayload() { }
+
+		public FilterType FilterType
+		{
+			get => (FilterType)_filterType;
+			internal set => _filterType = (byte)value;
+		}
+
+		public void ReadWrite(BitcoinStream stream)
+		{
+			stream.ReadWrite(ref _filterType);
+			stream.ReadWrite(ref _stopHash);
+			stream.ReadWrite(ref _previousFilterHeader);
+			stream.ReadWrite(ref _filterHashesLength);
+			var _tempfilterHeaders = new byte[_filterHashesLength.ToLong() * 32];
+
+			stream.ReadWrite(ref _tempfilterHeaders);
```

Comment:
if you are serializing you need to fill out tempfilterHeaders.

Reply:
Just to make sure I understand what you mean, when serializing instead of reading the tempfilterHeaders I want to write the current hashes to it?

Reply:
Meaning when serializing, I want to convert the array of uint256 hashes to a byte[] that can be written back into the network?

Reply:
I think the serialization issue is fixed now.